folding-schemes/src/folding/circuits/cyclefold.rs

```diff
@@ -37,8 +37,13 @@ use crate::folding::nova::{nifs::NIFS, CommittedInstance, Witness};
 use crate::frontend::FCircuit;
 use crate::Error;
 
-/// Public inputs length for the CycleFoldCircuit: |[r, p1.x,y, p2.x,y, p3.x,y]|
-pub(crate) const CF_IO_LEN: usize = 7;
+/// Public inputs length for the CycleFoldCircuit:
+/// For Nova this is: |[r, p1.x,y, p2.x,y, p3.x,y]|
+/// In general, |[r * (n_points-1), (p_i.x,y)*n_points, p_folded.x,y]|, thus, io len is:
+/// (n_points-1) + 2*n_points + 2
+pub fn cf_io_len(n_points: usize) -> usize {
+    (n_points - 1) + 2 * n_points + 2
+}
```

Comment:
This is kind of the replacement of the old `CF_IO_LEN` which was a constant, but now it depends on the number of points (which for example in HyperNova it comes from $\mu, \nu$ (the number of LCCCS&CCCS instances being folded per step in HyperNova)), that's why I kept it as a standalone helper function.

Regarding moving it to be a method of the `CycleFold` struct, there are multiple different `CycleFoldXXX` structs (`CycleFoldCircuit, ChallengeGadget, CycleFoldCommittedInstanceVar, ...`)
this method is agnostic to them, and it's just a method to help to obtain the number of public inputs based on the $\mu, \nu$ parameters.

We could think of adding it as a method for the `CycleFoldCircuit`, but for example it is used in the Nova's [AugmentedFCircuit](https://github.com/privacy-scaling-explorations/sonobe/blob/6730bf6e56c606095c3f5ebe9afb591e99c71914/folding-schemes/src/folding/nova/circuits.rs#L341), and this circuit does not use at all the `CycleFoldCircuit`, just the `cf_io_len` method to know how many inputs for the CommittedInstance, so it would just change the code from
```rust
        let cf_u_dummy = CommittedInstance::dummy(cf_io_len(self.mu + self.nu));
```
to
```rust
        let cf_u_dummy = CommittedInstance::dummy(CycleFoldCircuit::<C1, GC1>::cf_io_len(self.mu + self.nu));
```
which is the same but with a longer line of code, and additionally this won't compile since in the AugmentedFCircuit of Nova the `GC1` is not defined (since it is not needed).
So to move it as a method to the `CycleFoldCircuit` it would require adding to the `AugmentedFCircuit` of Nova another type to the impl (`GC1`) which is not really needed in that struct (nor in the method `generate_constraints` where having the `cf_io_len` as method of the `CycleFoldCircuit` would make it a requirement.

How do you see it?